lib/interface_quda.cpp

```diff
@@ -5960,3 +6026,143 @@ void gaugeObservablesQuda(QudaGaugeObservableParam *param)
   gaugeObservables(*gauge, *param, profileGaugeObs);
   profileGaugeObs.TPSTOP(QUDA_PROFILE_TOTAL);
 }
+
+void blasGEMMQuda(void *arrayA, void *arrayB, void *arrayC, bool use_native, QudaBLASParam *blas_param)
```

Comment:
At some point we'll need to refactor interface_quda.cpp since it's getting rather unwieldy.  We could start that now: perhaps move this function to its own file, blas_gemm_quda.cpp?  Perhaps leaving this stub here and pass the `profileBLAS` as a parameter to the function in that file?

Reply:
I support splitting up the `interface_quda.cpp` file, but if we are let's stick them in a `lib/interface/` subdirectory or something of the sort---start clawing back at the file sprawl 🙂

Reply:
Excellent idea!

Reply:
I'm afraid I'm going to need some help on this. I moved the `blasGEMMQuda` function to its own file `lib/interfaces/blas_interface.cpp`, constructed a `CMakeLists.txt`, added the sub directory, etc... Rather than use an `extern` declaration for the BLAS `TimeProfile` I created a simple method to return a reference to it in `interface_quda.cpp`:
```C++
//!< Profiler for GEMM and other BLAS                                                                                                                                             
static TimeProfile profileBLAS("blasQuda");
TimeProfile* getProfileBLAS() {
  return &profileBLAS;
}
```

which naturally must have a forward declaration in `quda.h`. But `quda.h` has no knowledge of what a `TimeProfile` object is. I don't know how to proceed.

Also, the call to `checkBLASParam(blas_param);` now causes an error, because the function is defined in `check_params.h`  which is no longer visible to the file in `lib/interfaces/blas_interface.cpp`, and the compilation of `interface_quda.cpp` causes an unused-function warning because there is no instance of `checkBLASParam()` in the file anymore.

Are we sure that this is in the scope of this PR, or am I missing an obvious solution?

Reply:
Nevermind... All fixed, thank you @maddyscientist .